### PR TITLE
Run pytest unit tests on circleci

### PR DESCRIPTION
**Pull request by rtburns-jpl**
This puts the circleci job at parity with the github action, so the github action is also removed in this PR.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -18,6 +18,14 @@ jobs:
           name: "Download S1 test data"
           command: |
             python3 tests/download_data.py
```
Comment (yunjunz) [inline .circleci/config.yml:20]: This downloading of real S1 data is taking a long time, and is currently not used anywhere. How about commenting it out for now?

Comment (rtburns-jpl) [reply .circleci/config.yml]: Sounds good, I disabled it.

**Review by yunjunz — APPROVED** [review]
LGTM. Thank you @rtburns-jpl.

**Review by LiangJYu — APPROVED** [review]
LGTM. Nice work!

**Pull request merged**